govcd/edgegateway.go

```diff
@@ -311,87 +339,172 @@ func (eGW *EdgeGateway) AddSNATRule(networkHref, externalIP, internalIP, descrip
 		return nil, fmt.Errorf("%s", combinedTaskErrorMessage(task.Task, err))
 	}
 
+	var createdNatRule *types.NatRule
+
 	err = eGW.Refresh()
 	if err != nil {
 		return nil, fmt.Errorf("error refreshing edge gateway: %#v", err)
 	}
 
-	return eGW.EdgeGateway.Configuration.EdgeGatewayServiceConfiguration.NatService.NatRule, nil
+	for _, natRule := range eGW.EdgeGateway.Configuration.EdgeGatewayServiceConfiguration.NatService.NatRule {
+		if natRule.Description == mappingId {
+			createdNatRule = natRule
+			break
+		}
+	}
+
+	if createdNatRule == nil {
+		return nil, fmt.Errorf("error creating SNAT rule, didn't match created rule")
+	}
+
+	createdNatRule.Description = description
+
+	return eGW.UpdateNatRule(createdNatRule)
+}
+
+// getPseudoUuid creates unique ID/UUID
+func getPseudoUuid() (string, error) {
+
+	b := make([]byte, 16)
+	_, err := rand.Read(b)
+	if err != nil {
+		return "", err
+	}
+
+	uuid := fmt.Sprintf("%X-%X-%X-%X-%X", b[0:4], b[4:6], b[6:8], b[8:10], b[10:])
+
+	return uuid, nil
+}
+
+// UpdateNatRule updates NAT rule and handles task. Returns updated NAT rule or error.
+func (eGW *EdgeGateway) UpdateNatRule(natRule *types.NatRule) (*types.NatRule, error) {
+	task, err := eGW.UpdateNatRuleAsync(natRule)
+	if err != nil {
+		return nil, fmt.Errorf("error updating NAT rule: %#v", err)
+	}
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return nil, fmt.Errorf("%s", combinedTaskErrorMessage(task.Task, err))
+	}
+
+	return eGW.GetNatRule(natRule.ID)
+}
+
+// UpdateNatRuleAsync updates NAT rule and returns task or error.
+func (eGW *EdgeGateway) UpdateNatRuleAsync(natRule *types.NatRule) (Task, error) {
+	if "" != natRule.GatewayNatRule.Protocol && !isValidProtocol(natRule.GatewayNatRule.Protocol) {
```

Comment:
Please change `if "" != natRule.GatewayNatRule.Protocol` to `if natRule.GatewayNatRule.Protocol != ""`

Reply:
Fixed